src/eca/handlers.clj

```diff
@@ -131,4 +133,43 @@
    :eca/mcp-start-server
    (f.tools/start-server! (:name params) db* messenger config)))
 
-(defn chat-selected-behavior-changed [{:keys []} {:keys [_behavior]}])
+(defn ^:private update-behavior-model!
+  "Updates the selected model based on behavior configuration."
+  [behavior-config config messenger db*]
+  (when-let [model (or (:defaultModel behavior-config)
+                       (:defaultModel config))]
+    (config/notify-fields-changed-only!
+     {:chat {:select-model model}}
+     messenger
+     db*)))
+
+(defn ^:private update-tool-servers!
```

Comment:
we may want to have this function in `f.tool/refresh-tool-servers` or something.